### PR TITLE
export Meta.isidentifier (and others)

### PR DESCRIPTION
**Pull request by vtjnash**
May be useful for embedding into other languages, for example.

Rebased and extended from https://github.com/JuliaLang/julia/pull/13283 (marked forget-me-not) 
Co-authored-by: Ivar Nesje <ivarne@gmail.com>
Co-authored-by: Jameson Nash <vtjnash@gmail.com>

### CODE REVIEW

File: base/show.jl [hunk 1]
```diff
@@ -1131,6 +1131,26 @@ const expr_parens = Dict(:tuple=>('(',')'), :vcat=>('[',']'),
 
 is_id_start_char(c::AbstractChar) = ccall(:jl_id_start_char, Cint, (UInt32,), c) != 0
 is_id_char(c::AbstractChar) = ccall(:jl_id_char, Cint, (UInt32,), c) != 0
+
+"""
+     isidentifier(s) -> Bool
+
+Returns whether the symbol or string `s` contains characters that are parsed as
```
Comment (rfourquet) [inline base/show.jl:1138]:
```suggestion
Return whether the symbol or string `s` contains characters that are parsed as
```

File: base/show.jl [hunk 2]
```diff
@@ -1131,6 +1131,26 @@ const expr_parens = Dict(:tuple=>('(',')'), :vcat=>('[',']'),
...
+Internally Julia allows any sequence of characters in a `Symbol` (except `\0`s),
+and macros automatically use variable names containing `#` in order to avoid
+naming collision with the surrounding code. In order for the parser to
+recognize a variable, it uses a limited set of characters (greatly extended by
+Unicode). `isidentifier()` makes it possible to query the parser directly
+whether a symbol contains valid characters.
+
+# Examples
+```jldoctest
+julia> Meta.isidentifier(:x), Meta.isidentifier("1x")
```
Comment (rfourquet) [inline base/show.jl:1150]:
```suggestion
julia> isidentifier(:x), isidentifier("1x")
```

**Review by JeffBezanson — APPROVED** [review]

File: base/show.jl (hunk 2 above)
Comment (vtjnash) [reply base/show.jl:1150]: They won't be available at the REPL natively

Comment (rfourquet) [reply base/show.jl:1150]: I thought it would, as you seem to export it from Base.

File: base/exports.jl [hunk 3]
```diff
@@ -278,6 +278,7 @@ export
     isapprox,
     iseven,
     isfinite,
+    isidentifier,
```
Comment (vtjnash) [inline base/exports.jl:281]:
```suggestion
```

File: base/show.jl (hunk 2 above)
Comment (vtjnash) [reply base/show.jl:1150]: ah, whoops, I forgot to drop that file from the commit

**Pull request merged**